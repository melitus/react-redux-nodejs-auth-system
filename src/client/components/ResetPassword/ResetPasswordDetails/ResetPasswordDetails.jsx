// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';

import { Input } from '../../Field/index';

const ResetPasswordDetails = ({ password, confirmPassword }) => (
  <FormSection name="resetPasswordDetails">
    <label target="resetPasswordForm.resetPasswordDetails.password">Password</label>
    <Field
      id="resetPasswordForm.resetPasswordDetails.password"
      name="password"
      component={Input}
      placeholder="Password"
      type="password"
      value={password}
    />

    <label target="resetPasswordForm.resetPasswordDetails.confirmPassword">Confirm Password</label>
    <Field
      id="resetPasswordForm.resetPasswordDetails.confirmPassword"
      name=" confirmPassword"
      component={Input}
      placeholder="Confirm Password"
      type="password"
      value={ confirmPassword}
    />
  </FormSection>
);

ResetPasswordDetails.defaultProps = {
  email: '',
   confirmPassword: '',
};

export default ResetPasswordDetails;
