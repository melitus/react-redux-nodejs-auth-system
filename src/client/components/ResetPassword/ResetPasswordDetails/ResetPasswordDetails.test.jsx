import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import ResetPasswordDetails from './ResetPasswordDetails';

enzyme.configure({ adapter: new Adapter() });

describe('<ResetPasswordDetails />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <ResetPasswordDetails />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
