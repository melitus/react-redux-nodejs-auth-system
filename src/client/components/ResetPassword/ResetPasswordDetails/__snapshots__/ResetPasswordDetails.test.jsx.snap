exports[`<ResetPasswordDetails /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPasswordDetails
    confirmPassword=""
    email="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="resetPasswordForm.resetPasswordDetails.password">
          Password
</label>,
        <Field
          component={undefined}
          id="resetPasswordForm.resetPasswordDetails.password"
          name="password"
          placeholder="Password"
          type="password"
          value={undefined} />,
        <label
          target="resetPasswordForm.resetPasswordDetails.confirmPassword">
          Confirm Password
</label>,
        <Field
          component={undefined}
          id="resetPasswordForm.resetPasswordDetails.confirmPassword"
          name=" confirmPassword"
          placeholder="Confirm Password"
          type="password"
          value="" />,
      ],
      "component": "div",
      "name": "resetPasswordDetails",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password",
          "target": "resetPasswordForm.resetPasswordDetails.password",
        },
        "ref": null,
        "rendered": "Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "resetPasswordForm.resetPasswordDetails.password",
          "name": "password",
          "placeholder": "Password",
          "type": "password",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Confirm Password",
          "target": "resetPasswordForm.resetPasswordDetails.confirmPassword",
        },
        "ref": null,
        "rendered": "Confirm Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "resetPasswordForm.resetPasswordDetails.confirmPassword",
          "name": " confirmPassword",
          "placeholder": "Confirm Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="resetPasswordForm.resetPasswordDetails.password">
            Password
</label>,
          <Field
            component={undefined}
            id="resetPasswordForm.resetPasswordDetails.password"
            name="password"
            placeholder="Password"
            type="password"
            value={undefined} />,
          <label
            target="resetPasswordForm.resetPasswordDetails.confirmPassword">
            Confirm Password
</label>,
          <Field
            component={undefined}
            id="resetPasswordForm.resetPasswordDetails.confirmPassword"
            name=" confirmPassword"
            placeholder="Confirm Password"
            type="password"
            value="" />,
        ],
        "component": "div",
        "name": "resetPasswordDetails",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password",
            "target": "resetPasswordForm.resetPasswordDetails.password",
          },
          "ref": null,
          "rendered": "Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "resetPasswordForm.resetPasswordDetails.password",
            "name": "password",
            "placeholder": "Password",
            "type": "password",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirm Password",
            "target": "resetPasswordForm.resetPasswordDetails.confirmPassword",
          },
          "ref": null,
          "rendered": "Confirm Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "resetPasswordForm.resetPasswordDetails.confirmPassword",
            "name": " confirmPassword",
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
